journey-map/public/index.mjs

```diff
@@ -1,204 +1,120 @@
-// This allows us to skip
-const renderTransitions = (journeyMap, formData) => {
-    const states = [...initialStates];
-    const stateTransitions = [];
-
-    for (let i = 0; i < states.length; i++) {
-        const state = states[i];
-        const definition = journeyMap[state];
-        const events = definition.events || definition.exitEvents || {};
-
-        const eventsByTarget = {};
-        Object.entries(events).forEach(([eventName, def]) => {
-            const target = resolveEventTarget(def, formData);
-
-            if (errorStates.includes(target) && !formData.has('includeErrors')) {
-                return;
-            }
-            if (failureStates.includes(target) && !formData.has('includeFailures')) {
-                return;
-            }
-
-            if (!states.includes(target)) {
-                states.push(target);
-            }
-            eventsByTarget[target] = eventsByTarget[target] || [];
-            eventsByTarget[target].push(eventName);
-        });
-
-        Object.entries(eventsByTarget).forEach(([target, eventNames]) => {
-            stateTransitions.push(`    ${state}-->|${eventNames.join('\\n')}|${target}`);
-        });
+// Render the journey map SVG
+const renderSvg = async () => {
+    const diagram = render(journeyMap, nestedJourneys, new FormData(form));
+    const diagramElement = document.getElementById('diagram');
+    const { svg, bindFunctions } = await mermaid.render('diagramSvg', diagram);
+    diagramElement.innerHTML = svg;
+    if (bindFunctions) {
+        bindFunctions(diagramElement);
     }
-
-    return { transitionsMermaid: stateTransitions.join('\n'), states };
+    svgPanZoom('#diagramSvg');
 };
 
-const renderStates = (journeyMap, states) => {
-    // Types
-    // process - response.type = process, response.lambda = <lambda>
-    // page    - response.type = page, response.pageId = 'page-id'
-    // cri     - response.type = cri,
-    const mermaids = states.map((state) => {
-        const definition = journeyMap[state];
+const highlightState = (state) => {
+    // Remove existing highlights
+    Array.from(document.getElementsByClassName('highlight'))
+        .forEach((edge) => edge.classList.remove('highlight', 'outgoingEdge', 'incomingEdge'));
+
+    // Add new highlights
+    Array.from(document.getElementsByClassName(`LS-${state}`))
+        .forEach((edge) => edge.classList.add('highlight', 'outgoingEdge'));
+    Array.from(document.getElementsByClassName(`LE-${state}`))
+        .forEach((edge) => edge.classList.add('highlight', 'incomingEdge'));
+    Array.from(document.getElementsByClassName('node'))
+        .filter((node) => node.id.startsWith(`flowchart-${state}-`))
+        .forEach((node) => node.classList.add('highlight'));
+};
 
-        switch (definition.response?.type) {
+// Set up the click handlers that mermaid binds to each node
+const setupMermaidClickHandlers = () => {
+    const getDesc = (def) => {
```

Comment:
I don't object to it, but why define the `getDesc` function within `setupMermaidClickHandlers`? Is it a way to effectively make it "private"?

Reply:
Yeah, there's no good reason really except that it felt specific to this method rather than a general thing. I could happily pull it out, but probably won't for now!